proto/gnmi_ext/gnmi_ext.proto

```diff
@@ -89,3 +92,47 @@ message TimeRange {
   int64 start = 1; // Nanoseconds since the epoch
   int64 end = 2;   // Nanoseconds since the epoch
 }
+
+// The CommitID is returned by the gNMI server to indicate the ID of a commit.
+// It is returned in response to the Set Request with the CommitConfirm message
+// that had COMMIT_CONFIRMED_ACTION_START set.
+// The CommitID is used in the subsequent SetRequest/CommitConfirm messages
+// to identify commit to be confirmed, reset or rejected.
+message CommitID {
+  string id = 1;
+}
+
+// The CommitConfirmed allows automatic rollback of a commit
+// if Set changes are not confirmed within the specified timeout.
+// The document about gNMI commit confirmed can be found at
+// https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-commit-confirmed.md
+message CommitConfirmed {
+  CommitConfirmedAction action = 1;
```

Comment:
What is the rational behind defining _action_ as an enum over what is proposed in https://github.com/openconfig/reference/issues/197 which is to use `oneof id` which implicitly models the 3 actions - commit/confirm/cancel(START/CONFIRM/REJECT) and also implicitly enforces required behavior at the proto level, mandating the request to be only one of these actions at a time.

Reply:
To eliminate confusion of the spec appearing to use different IDs - I have updated the `oneof` to represent `actions` instead of IDs - https://github.com/openconfig/gnmi/pull/155/files. With this change both the proposal looks identical in terms of semantics and functionality.

I believe that using `oneof` over `ENUM` for this particular spec helps define actions much more clearly. Eg. `rollback_duration` is only valid during commit. `oneof` enforces that behavior, with ENUMs client can pass `rollback_duration` field during confirm or cancel request. Its best to avoid unnecessary explanation in documentation  if we can enforce them in the spec which makes the API much cleaner.

Reply:
@kidiyoor @dplore I think one of the fundamental differences between #154 and #155 is that in our proposal, the `CommitID` message is returned by the gNMI server. In other words, the server assigns Commit IDs and not the client selecting them.

This matched our implementation on both platforms (SR Linux and SR OS), and I feel like this is also true for most other NOSes that currently support commit operations.